encoding/thrift/outbound.go

```diff
@@ -143,14 +190,15 @@ func (c thriftClient) Call(ctx context.Context, reqMeta yarpc.CallReqMeta, reqBo
 	if err != nil {
 		// ToWire validates the request. If it failed, we should return the error
 		// as-is because it's not an encoding error.
-		return wire.Value{}, nil, err
+		return nil, nil, err
 	}
 
 	reqEnvelopeType := reqBody.EnvelopeType()
-	if reqEnvelopeType != wire.Call {
-		return wire.Value{}, nil, encoding.RequestBodyEncodeError(
-			&treq, errUnexpectedEnvelopeType(reqEnvelopeType),
-		)
+	if reqEnvelopeType != wire.Call && reqEnvelopeType != wire.OneWay {
+		return nil, nil,
+			encoding.RequestBodyEncodeError(
+				&treq, errUnexpectedEnvelopeType(reqEnvelopeType),
```

Comment:
nit: weird spacing. pull the call onto the previous line. The arguments can be
on the next line.

```go
return nil, nil, encoding.RequestBodyEncodeError(
    &treq, errUnexpectedEnvelopeType(reqEnvelopeType),
)
```